From dd3854a2416fcb381af7b40eab132826c1ea6fe2 Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Wed, 14 Nov 2018 14:55:55 +0200
Subject: [PATCH 5138/5242] MLK-20216-4: video/fbdev/imx_dcss: remove useless
 function call

commit  80f016350a481638c492876ce300113ed9c5c5e7 from
https://source.codeaurora.org/external/imx/linux-imx.git

This fixes Coverity issue since the function does nothing. The function
only returns some values which are never used.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
CC: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/imx_dcss.c |   25 -------------------------
 1 file changed, 25 deletions(-)

diff --git a/drivers/video/fbdev/mxc/imx_dcss.c b/drivers/video/fbdev/mxc/imx_dcss.c
index 0da21d4..d240331 100644
--- a/drivers/video/fbdev/mxc/imx_dcss.c
+++ b/drivers/video/fbdev/mxc/imx_dcss.c
@@ -1211,30 +1211,6 @@ static int dcss_clks_rate_set(struct dcss_info *info)
 	return 0;
 }
 
-static int dcss_dec400d_config(struct dcss_info *info,
-			       bool decompress, bool resolve)
-{
-	struct dcss_channel_info *chan_info;
-	struct cbuffer *cb;
-
-	if (resolve == true)
-		return -EINVAL;
-
-	/* dec400d always in channel 1 */
-	chan_info = &info->chans.chan_info[0];
-	cb = &chan_info->cb;
-
-	if (decompress == true) {
-		/* TODO: configure decompress */
-		;
-	} else {
-		/* TODO: configure bypass */
-		;
-	}
-
-	return 0;
-}
-
 static int dcss_dtrc_config(uint32_t dtrc_ch,
 			    struct dcss_info *info,
 			    bool decompress,
@@ -1311,7 +1287,6 @@ static int dcss_decomp_config(uint32_t decomp_ch, struct dcss_info *info)
 
 	switch (decomp_ch) {
 	case 0:		/* DEC400D */
-		dcss_dec400d_config(info, need_decomp, need_resolve);
 		break;
 	case 1:		/* DTRC1   */
 	case 2:		/* DTRC2   */
-- 
1.7.9.5

